mangaki/mangaki/utils/recommendations.py

```diff
@@ -87,18 +87,16 @@ def get_reco_algo(request, algo_name='knn', category='all'):
                 for work_id, choice in current_user_ratings(request).items()
                 if choice not in ('willsee', 'wontsee')
             ])
-    else:
-        # Every rating is useful
-        triplets = list(
-            Rating.objects.values_list('user_id', 'work_id', 'choice'))
-
-    chrono.save('get all %d interesting ratings' % len(triplets))
 
-    dataset = Dataset()
     try:
         algo = get_algo_backup(algo_name)
         dataset = get_dataset_backup(algo_name)
     except FileNotFoundError:
+        # Every rating is useful
```

Comment:
Indeed, simple fix: I could check if `triplets` exists already, as KNN set it upfront, do you prefer to wait for @jilljenn input or take this discussion into another issue?